src/@types/casperlabsSigner.d.ts

```diff
@@ -15,12 +15,13 @@ interface CasperLabsHelper {
   requestConnection: () => void;
   
   /**
-   * send base16 encoded message to plugin to sign
+   * Send Deploy in JSON format message to Signer plugin to sign.
    *
-   * @param messageBase16 the base16 encoded message that plugin received to sign
-   * @param publicKeyBase64 the base64 encoded public key used to sign the deploy, if set, we will check whether it is the same as the active key for signing the message, otherwise, we won't check.
+   * @param deploy deploy in JSON format
+   * @param sourcePublicKey base64 encoded public key used to sign the deploy
+   * @param targetPublicKey base64 encoded public key used to sign the deploy
    */
-  sign: (messageBase16: string, publicKeyBase64?: string) => Promise<string>;
+  sign: (deploy: JSON, sourcePublicKey: string, targetPublicKey: string) => Promise<JSON>;
```

Comment:
To help cement the idea that we want them to use hex format would it be worth renaming these to:
`sourcePublicKeyHex` and `targetPublicKeyHex`?
Just opinion on this one as the doc will state it explicitly once updated.

Also I can't quite remember but wasn't there an issue with typing `deploy` as `JSON` - I feel like it was set to `JSONTypes`